app/alias_utils.py

```diff
@@ -27,10 +32,131 @@
     Mailbox,
     EmailLog,
     Contact,
+    AutoCreateRule,
 )
 from app.regex_utils import regex_match
 
 
+def get_user_if_alias_would_auto_create(
+    address: str, notify_user: bool = False
+) -> Optional[User]:
+    banned_prefix = f"{VERP_PREFIX}."
+    if address.startswith(banned_prefix):
+        LOG.w("alias %s can't start with %s", address, banned_prefix)
+        return None
+
+    try:
+        # Prevent addresses with unicode characters (🤯) in them for now.
+        validate_email(address, check_deliverability=False, allow_smtputf8=False)
+    except EmailNotValidError:
+        return None
+
+    will_create = check_if_alias_can_be_auto_created_for_custom_domain(
```

Comment:
it doesn't return a `CustomDomain` it returns either `None` or `Tuple[CustomDomain,Optional[AutoCreateRule]]` . To prevent confusion I woudn't call it `custom_domain`. Maybe domain_with_rule?

Reply:
Renamed to `domain_and_rule`